tests/test_language_detect.py

```diff
@@ -0,0 +1,84 @@
+from parameterized import parameterized, param
+from datetime import datetime
+
+from tests import BaseTestCase
+
+from dateparser.custom_lang_detect.fast_text import detect_languages as fast_text
+from dateparser.custom_lang_detect.lang_detect import detect_languages as lang_detect
+from dateparser import parse
+from dateparser.date import DateDataParser
+from dateparser.search import search_dates
+
+class CustomLangDetectParser(BaseTestCase):
+    def setUp(self):
+        super().setUp()
+        self.result = NotImplemented
+        self.date_string = NotImplemented
+        self.parser = NotImplemented
+        self.translated = NotImplemented
+
+    def check_is_returned_list(self):
+        self.assertEqual(type(self.result), type([]))
+
+    # Unit test for internally supported fasttext and langdetect.
+
+    @parameterized.expand([
+        param(dt_string="14 June 2020"),
+    ])
+    def test_custom_language_detect_fast_text(self, dt_string):
+        self.result = fast_text(dt_string)
+        self.check_is_returned_list()
+
+    @parameterized.expand([
+        param(dt_string="14 June 2020"),
+    ])
+    def test_custom_language_detect_lang_detect(self, dt_string):
+        self.result = lang_detect(dt_string)
+        self.check_is_returned_list()
+
+    # Mock test for parse, search_dates and DateDataParser
+
+    # parse
+
+    def when_date_is_parsed_using_fast_text_parse(self, dt_string):
+        self.result = parse(dt_string, detect_languages_func=fast_text)
```

Comment:
the idea is good, but this is not a "mock". What you can do is mocking the fasttext library.
Here an example:  https://github.com/scrapinghub/dateparser/blob/803d4451ccc18e147363b7824cd38db4307e0740/tests/test_date.py#L522-L527

In the case of `langdetect`, for example, you could mock `langdetect.detect_langs` to return whatever you want.

@lopuhin, what do you think? Any suggestion when testing/mocking?

Reply:
+1, I think we can mock the language detection library entry-point

Reply:
Is the implemented mock correct?

Pl suggest